components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/sequence/impl/GraphBasedSequenceHandler.java

```diff
@@ -139,6 +142,27 @@ private void handleEndOfSequence(HttpServletRequest request, HttpServletResponse
         }
     }
 
+    /**
+     * Process FailNode
+     * @param request
+     * @param response
+     * @param context
+     * @param sequenceConfig
+     * @param node
+     * @throws FrameworkException
+     */
+    private void handleAuthFail(HttpServletRequest request, HttpServletResponse response,
+                                AuthenticationContext context, SequenceConfig sequenceConfig, AuthGraphNode node)
+            throws FrameworkException {
+
+        if (log.isDebugEnabled()) {
+            log.debug("There are no more steps to execute");
```

Comment:
Better the comment be
"The current node is a \"Fail\" node. Dynamic scripting has decided that authentication id failed."